packages/ui/src/components/TimeInputV2/index.tsx

```diff
@@ -0,0 +1,511 @@
+    }
+
+    &:not([data-disabled="true"]):not([data-readonly="true"]):hover {
+      border-color: ${({ theme }) => theme.colors.danger.borderHover};
+    }
+  }
+`
+
+export const Input = styled.input<{
+  'data-size': 'small' | 'medium' | 'large'
+  'data-period'?: boolean
+}>`
+  border: none;
+  outline: none;
+  background: transparent;
+  font-size: ${({ theme }) => theme.typography.bodySmall.fontSize};
+  width: ${({ theme }) => theme.sizing[312]};
+  height: ${({ theme }) => theme.sizing[300]};
+  text-align: center;
+  border-radius: ${({ theme }) => theme.radii.default};
+  color: ${({ theme }) => theme.colors.neutral.text};
+  caret-color: transparent;
+
+  &[data-size='large'] {
+    font-size: ${({ theme }) => theme.typography.body.fontSize};
+  }
+
+  &:not(:disabled):hover {
+    background-color: ${({ theme }) => theme.colors.neutral.backgroundHover};
+    color: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
+
+  &:not(:disabled):active, 
+  :not(:disabled):focus{
+    background-color: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+    color:  ${({ theme }) => theme.colors.neutral.text};
+  }
+
+  &:read-only {
+    cursor: default;
+  }
+
+  &:disabled {
+    cursor: not-allowed;
+    user-select: none;
+  }
+
+  &[data-period="true"] {
+    color: ${({ theme }) => theme.colors.neutral.textWeak};
+  }
+
+  ::-moz-selection {
+    background: none;
+  }
+
+  ::selection {
+    background: none;
+  }
+`
+
+const CustomText = styled(Text)`
+padding-inline: ${({ theme }) => theme.space['0.25']};
+`
+
+type TimeInputProps = {
+  placeholder?: Time
+  value?: Time
+  clearable?: boolean
+  required?: boolean
+  labelDescription?: ReactNode
+  helper?: ReactNode
+  disabled?: boolean
+  readOnly?: boolean
+  error?: boolean | string
+  'data-testid'?: string
+  onChange?: (value: Time, valuePeriod?: string) => void
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void
+  onFocus?: (event: FocusEvent<HTMLInputElement>) => void
+  className?: string
+  id?: string
+  size?: 'small' | 'medium' | 'large'
+  timeFormat?: 12 | 24
+  /**
+   * Automatically focus on the element on render. Autofocus is applied to the hour input
+   */
+  autoFocus?: boolean
+} & LabelProp
+
+/**
+ * A time input component that allows users to type a time in a 24 or 12-hour format.
+ * @experimental This component is experimental and may be subject to breaking changes in the future.
+ */
+export const TimeInputV2 = ({
+  label,
+  timeFormat = 24,
+  value,
+  clearable,
+  required,
+  labelDescription,
+  helper,
+  size = 'medium',
+  disabled = false,
+  readOnly = false,
+  error = false,
+  onChange,
+  onBlur,
+  onFocus,
+  className,
+  id,
+  autoFocus,
+  'data-testid': dataTestId,
+  placeholder = DEFAULT_PLACEHOLDER,
+}: TimeInputProps) => {
+  const correctEmpty = timeFormat === 24 ? EMPTY_TIME_24 : EMPTY_TIME_12
+  const [time, setTime] = useState<Time>(
+    value ? formatValue(value, timeFormat) : correctEmpty,
+  )
```

Comment:
There is an issue for in the value. If I look at the old time input we use to have a date that includes the time into it. This is the right way to do it as we want to use the standard date type for it. Here we use a custom thing and it differ with our old usage and does not take in account all the specificity of a time (time zone, etc)

![Screenshot 2025-01-09 at 11 47 36](https://github.com/user-attachments/assets/8ff62d31-a2d1-47d4-855e-2a560bab0dd6)
![Screenshot 2025-01-09 at 11 47 47](https://github.com/user-attachments/assets/bca39e19-adfd-4444-956a-a8f8e1e7b6a8)

Reply:
I guess each time we change the time you could create a `new Date()` with the current date and time set on the input